### PR TITLE
Update ML Workspace to latest API

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Update ML workspace to latest api
- Added additional features
- Added min test

| Pipeline |
| - |
| [MachineLearningServices: Workspaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.machinelearningservices.workspaces.yml)|

# Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: arm/Microsoft.MachineLearningServices/workspaces/.parameters/min.parameters.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        "associatedStorageAccountResourceId": {
+            "value": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Storage/storageAccounts/adp<<namePrefix>>azsax001"
+        },
+        "associatedKeyVaultResourceId": {
+            "value": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.KeyVault/vaults/adp-<<namePrefix>>-az-kv-x-001"
+        },
+        "associatedApplicationInsightsResourceId": {
+            "value": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Insights/components/adp-<<namePrefix>>-az-appi-x-001"
+        },
+        "systemAssignedIdentity": {
```
Comment (eriqua) [inline arm/Microsoft.MachineLearningServices/workspaces/.parameters/min.parameters.json:20]: I'd suggest to remove it since it's not a required parameter

Comment (MrMCake) [reply arm/Microsoft.MachineLearningServices/workspaces/.parameters/min.parameters.json:20]: I think I had to add it as it failed without. Will retest

Comment (eriqua) [reply arm/Microsoft.MachineLearningServices/workspaces/.parameters/min.parameters.json]: In that case would the parameter be required/conditional?

Comment (MrMCake) [reply arm/Microsoft.MachineLearningServices/workspaces/.parameters/min.parameters.json]: image

Must be there in some capacity. Will update

Comment (MrMCake) [reply arm/Microsoft.MachineLearningServices/workspaces/.parameters/min.parameters.json:20]: @eriqua, it's true, either you need set `systemAssignedIdentity` to true, or set both `userAssignedIdentities` & `primaryUserAssignedIdentity`. Tested both succesfully

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**